File: app/src/main/java/com/somethingfun/jay/peanut/drawing/Line.java
package com.somethingfun.jay.peanut.drawing;

import android.graphics.Canvas;
import android.graphics.Paint;

import com.somethingfun.jay.peanut.drawing.DrawableObject;

/**
 * Created by jay on 17. 11. 10.
 */

public class Line extends DrawableObject {

    private float startX;
    private float startY;
    private float stopX;
    private float stopY;


    public Line(float startX, float startY, float stopX, float stopY) {
        this.startX = startX;
        this.startY = startY;
        this.stopX  = stopX;
        this.stopY  = stopY;

        paint = new Paint(Paint.ANTI_ALIAS_FLAG);
        paint.setStyle(Paint.Style.FILL);
        paint.setColor(0xff101010);
        paint.setStrokeWidth(10);
    }

    @Override
    protected void drawInitialState(Canvas canvas) {
        canvas.drawLine(startX, startY, stopX, stopY, paint);
    }

    @Override
    protected void drawLastState(Canvas canvas) {
        canvas.drawLine(startX, startY, stopX, stopY, paint);
    }

    @Override
    protected void drawInAnimation(Canvas canvas, float interpolation) {
        float deltaX = stopX - startX;
        float deltaY = stopY - startY;
        float toX = deltaX * interpolation + startX;
        float toY = deltaY * interpolation + startY;
        canvas.drawLine(startX, startY, toX, toY, paint);
    }
}

File: peanutview/src/main/java/com/jeongmin/peanutview/drawing/shape/Line.kt
package com.jeongmin.peanutview.drawing.shape

import android.graphics.PointF

/**
 * Created by jay on 17. 12. 24.
 */
class Line {
    var x1: Float
    var y1: Float
    var x2: Float
    var y2: Float

    constructor(x1: Float, y1: Float, x2: Float, y2: Float) {
        this.x1 = x1
        this.x2 = x2
        this.y1 = y1
        this.y2 = y2
    }

    constructor(p1: PointF, p2: PointF) {
        this.x1 = p1.x
        this.y1 = p1.y
        this.x2 = p2.x
        this.y2 = p2.y
    }
}